### PR TITLE
fix: go plugin config parse

### PR DESCRIPTION
**Pull request by wangfakang**
Commit Message: fix the wrong ASSERT.
Additional Description: fix: go plugin config parse
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by wangfakang** [comment]
dependency internal Server Error.

/retest

File: contrib/golang/filters/http/test/golang_integration_test.cc [hunk 1]
```diff
@@ -76,7 +75,7 @@ TEST_P(GolangIntegrationTest, Echo) {
   EXPECT_EQ("403", response->headers().getStatusValue());
 
   // check body for echo
-  auto body = StringUtil::toUpper(absl::StrFormat("forbidden from go, path: %s\r\n", path));
+  auto body = StringUtil::toUpper(absl::StrFormat("%s, path: %s\r\n", echo_body, path));
   EXPECT_EQ(body, StringUtil::toUpper(response->body()));
```
Comment (doujiang24) [inline contrib/golang/filters/http/test/golang_integration_test.cc:79]: `StringUtil::toUpper` is not necessary.

**Review by doujiang24 — COMMENTED** [review]
one comment, otherwise, LGTM.

File: contrib/golang/filters/http/test/golang_integration_test.cc (hunk 1 above)
Comment (wangfakang) [reply contrib/golang/filters/http/test/golang_integration_test.cc]: Yes, it could delete now. Previously, headers were case insensitive.

**Comment by phlax** [comment]
a very high-level comment - if you make the response configurable - ie it doesnt echo the request - then im wondering if it should still be called echo

**Review by doujiang24 — APPROVED** [review]

**Comment by wangfakang** [comment]
> a very high-level comment - if you make the response configurable - ie it doesnt echo the request - then im wondering if it should still be called echo

Thank you for your comments.
In the current configuration, the config's `body` is a part of the response body. So there will be a echo body anyway.

**Review by ggreenway — APPROVED** [review]

**Pull request merged**